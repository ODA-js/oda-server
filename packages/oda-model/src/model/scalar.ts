import {
  MetaModelType,
  ScalarInput,
  ScalarStorage,
  ScalarMeta,
} from './interfaces';
import { ModelBase } from './modelbase';

export class Scalar extends ModelBase<ScalarMeta, ScalarInput, ScalarStorage> {
  public modelType: MetaModelType = 'scalar';
}
