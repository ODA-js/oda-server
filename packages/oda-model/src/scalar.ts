import { merge } from 'lodash';
import { MetaModelType } from './model';
import {
  ModelBase,
  ModelBaseInput,
  ModelBaseInternal,
  IModelBase,
} from './modelbase';
import { ElementMetaInfo } from './element';

/**
 * scalar item
 */
export interface IScalar extends IModelBase<ScalarMetaInfo, ScalarInput> {}

export interface ScalarMetaInfo extends ElementMetaInfo {}

export interface ScalarInternal extends ModelBaseInternal<ScalarMetaInfo> {}

export interface ScalarInput extends ModelBaseInput<ScalarMetaInfo> {}

const defaultMetaInfo = {};
const defaultInternal = {};
const defaultInput = {};

export class Scalar
  extends ModelBase<ScalarMetaInfo, ScalarInput, ScalarInternal>
  implements IScalar {
  public modelType: MetaModelType = 'scalar';
  constructor(inp: ScalarInput) {
    super(merge({}, defaultInput, inp));
    this.metadata_ = merge({}, defaultMetaInfo, this.metadata_);
    this.$obj = merge({}, defaultInternal, this.$obj);
  }
}
