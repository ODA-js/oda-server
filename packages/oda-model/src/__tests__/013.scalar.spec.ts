import 'jest';
import { Scalar, scalarDefaultMetaInfo } from '../scalar';

describe('Scalar', () => {
  it('default', () => {
    const res = new Scalar({ name: 'JSON' });
    expect(res.metadata).toMatchObject(scalarDefaultMetaInfo);
    expect(res.toObject()).toMatchSnapshot('toObject');
  });
});
