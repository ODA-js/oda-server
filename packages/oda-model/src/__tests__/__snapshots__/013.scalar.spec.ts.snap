// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scalar default: toObject 1`] = `
Object {
  "description": undefined,
  "metadata": Object {},
  "name": "JSON",
  "title": undefined,
}
`;
